### PR TITLE
[webview_flutter] Fix Flaky Resize Test

### PR DESCRIPTION
**Pull request by bparrishMines**
Split of https://github.com/flutter/plugins/pull/4455

~~After reading through https://flutter.dev/docs/cookbook/testing/widget/introduction. It seems that this test isn't using a Widget test in the expected way.~~ After more testing, the current solution does work in updating the widget, but still has a false positive for the 2nd reason below.

~~1. It's attempting to change the size of a `WebView` by pumping a new `Widget`.~~
2. I was able to make this test fail by adding a  `await Future.delayed(Duration(seconds: 2));` before the `boolean` check. This shows that resize is called regardless of changing the width and height.

This PR follows the Flutter cookbook example of a Widget test and creates a widget that changes the WebView on a button press. It also waits for the initial resize of the WebView from the initial load.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart [hunk 1]
```diff
@@ -162,91 +163,24 @@ void main() {
...
-              width: 200,
-              height: 200,
-              child: webView,
-            ),
-          ],
-        ),
-      ),
-    );
+    ));
+    await tester.pump(Duration(seconds: 3));
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:175]: This is required to wait for the initial resize of the `WebView`. The line `await pageFinishedCompleter.future;` can finish before this.

Comment (stuartmorgan) [reply packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:175]: Please capture this in a code comment, as it's non-obvious.

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart [hunk 2]
```diff
@@ -162,91 +163,24 @@ void main() {
...
-              width: 400,
-              height: 400,
-              child: webView,
-            ),
-          ],
-        ),
-      ),
-    );
+    await tester.tap(find.byKey(const ValueKey('resizeButton')));
+    await tester.pump(Duration(seconds: 3));
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:181]: How were the 3-second values chosen? It seems like a very arbitrary number.

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits.

(Related though: do we need three copies of this test? It's not clear to me where exactly the functionality that it's intended to test lives; are there important parts that live in both native an app-facing-package-dart?)

File: packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart (hunk 2 above)
Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:181]: It is mostly arbitrarily chosen :/. I just chose a safe value to wait for. I considered just waiting for the first call to resize, but I wasn't 100% sure it gets called every time on every platform.

Comment (stuartmorgan) [reply packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart:184]: Wait, so it this always waiting 3 seconds? I'm not all that familiar with widget tests; I assumed this was a timeout. If it's a static 3 second wait, that's still going to be a source of flake.

Comment (bparrishMines) [reply packages/webview_flutter/webview_flutter/example/integration_test/webview_flutter_test.dart]: That's a good point. I changed the tests to wait for resize completers.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**